opm/simulators/flow/CpGridVanguard.hpp

```diff
@@ -255,9 +255,16 @@ class CpGridVanguard : public FlowBaseVanguard<TypeTag>
         // Check if input file contains Lgrs.
         //
         // If there are lgrs, create the grid with them, and update the leaf grid view.
-        if (const auto& lgrs = this->eclState().getLgrs(); lgrs.size() > 0) {
+        if (const auto& lgrs = this->eclState().getLgrs(); (lgrs.size() > 0) && (this->grid_->comm().size() == 1)) {
             OpmLog::info("\nAdding LGRs to the grid and updating its leaf grid view");
             this->addLgrsUpdateLeafView(lgrs, lgrs.size(), *this->grid_);
+
+            this->updateGridView_();
+            this->updateCellDepths_();
+            this->updateCellThickness_();
+        }
+        else {
+            Opm::OpmLog::warning("Adding LGRs in parallel run is not supported yet.\n");
```

Comment:
This will log a warning about adding LGRs in parallel if `lgrs.size() == 0` and that's probably not what we want.  We probably want something like
```C++
if (const auto& lgrs = ...) {
    // ...
}
else if (this->grid_->comm().size() > 1) {
    // ...
}
```
instead.  On a second note, we're already in the `Opm::` namespace here, so the call to `warning()` here should just be
```C++
OpmLog::warning(...);
```
instead of
```C++
Opm::OpmLog::warning(...);
```

Reply:
Good point, we don't want that warning if there are no LGRs. I'll fix it! Thanks!

Reply:
@bska, thanks again. Your suggestion is incorporated in OPM/opm-simulators#6248